framework/src/media/audio/audio_manager.c

```diff
@@ -274,13 +274,32 @@ static audio_manager_result_t find_audio_card(audio_io_direction_t direct)
 	return ret;
 }
 
+static audio_manager_result_t ioctl_with_path(const char *path, int request, unsigned long arg)
```

Comment:
1. rename is needed, audio_stream_device_control ??? please suggest nice name, ioctl is not good for middle layer's prefix.
2. instead of path, we should use id of card & device(card_info). you can use get_card_path.

Reply:
1. I think `control_audio_stream_device` is better, cos function should start with verb
2. 3. I'll fix it